src/EditorFeatures/Core/Implementation/IntelliSense/Completion/Controller.Session.cs

```diff
@@ -64,12 +64,19 @@ internal void UpdateModelTrackingSpan(SnapshotPoint initialPosition)
                 });
             }
 
-            public override void Stop()
+            public override void DismissEditorSession()
             {
                 AssertIsForeground();
                 this.PresenterSession.ItemSelected -= OnPresenterSessionItemSelected;
                 this.PresenterSession.ItemCommitted -= OnPresenterSessionItemCommitted;
-                base.Stop();
+                base.DismissEditorSession();
+            }
+
+            public void CommitEditorSession()
```

Comment:
So can only one of CommitEditorSession and DismissEditorSession be called?  If so, then why isn't CommitEditirSession also removing the event hookup for ItemSelected?

If both can be called, then do we have an issue with double deregistration of ItemCommitted?